BrainPortal/app/controllers/demands_controller.rb

```diff
@@ -0,0 +1,433 @@
+      symbolic_result, message, backtrace = approve_one(req)
+      puts symbolic_result.to_s
+      #message = "Account created and user notified" if message.blank? && symbolic_result == :all_ok
+      [ req, symbolic_result, message, backtrace ]
+    end
+
+    @results.compact!
+
+    render :action => :multi_action
+  end
+
+
+
+  def approve_one(req) #:nodoc:
+    result = nil
+    exception_trace = ""
+
+    # Trigger approval code
+    if req.respond_to?(:after_approval)
+      begin
+        result = req.after_approval
+      rescue => ex
+        exception_trace = "#{ex.class}: #{ex.message}\n" + ex.backtrace.join("\n")
+        return [ :failed_approval, 'ERROR: Exception when approving' , exception_trace ]
+      end
+    end
+
+    # Mark as approved
+    info           = result.diagnostics    rescue nil
+    plain_password = result.plain_password rescue nil
+
+    req.approved_by ||= current_user.login
+    req.approved_at ||= Time.now
+    req.save
+
+    # Notify user
+    if send_account_created_email(req,plain_password)
+      return [ :all_ok, info, nil ]
+    else
+      return [ :not_notifiable, 'ERROR: The User was created in CBRAIN, but the notification email failed to send.', nil ]
+    end
+
+
+  end
+
+
+
+  def resend_confirm #:nodoc:
+puts "IN CONTROLLER METHOD"
+    @demand = Demand.find(params[:id]) rescue nil
+
+    unless can_edit?(@demand)
+      redirect_to :action => :new
+      return
+    end
+
+    if send_confirm_email(@demand)
+      flash[:notice] = "A new confirmation email has been sent."
+    else
+      flash[:error] = "It seems some error occured. Email notification was probably not sent. Check your mailhost settings.\n"
+    end
+
+    sleep 1
+    redirect_to :action => :show, :id => @demand.id
+  end
+
+
+
+  private
+
+
+
+  def can_edit?(demand) #:nodoc:
+    return false if demand.blank?
+    return true  if demand[:session_id] == request.session_options[:id]
+    if !current_user.nil?
+      return true  if current_user.has_role?(:admin_user)
+    end
+    false
+  end
+
+
+
+  def send_confirm_email(demand) #:nodoc:
+    confirm_url = url_for(:controller => :demands, :action => :confirm, :id => demand.id, :only_path => false, :token => demand.confirm_token)
+    CbrainMailer.request_confirmation(demand, confirm_url).deliver
+    return true
+  rescue => ex
+    Rails.logger.error ex.to_s
+    #flash[:error] = "It seems some error occured. Email notification was probably not sent. Sorry. We'll look into that.\n"
+    return false
+  end
+
+
+
+  def send_account_created_email(demand, plain_password = nil) #:nodoc:
+    CbrainMailer.account_created(demand, plain_password).deliver
+    return true
+  rescue => ex
+    Rails.logger.error ex.to_s
+    #flash.now[:error] ||= "No email for records: "
+    #flash.now[:error]  += "#{demand.id} (#{ex.class.to_s}), "
+    return false
+  end
+
+
+
+  def send_admin_notification(demand) #:nodoc:
+    return if RemoteResource.current_resource.support_email
```

Comment:
I don't get this... no email is sent to admins if a support email is defined? And even if it's blank? Isn't this backwards?